### DISCUSSION TITLE
getBulk response received by machine, but not picked up by PySNMP

### DISCUSSION THREAD

**Discussion by daniel-fackrell-vasion** [pysnmp/pysnmp#42] [General]
I'm currently running 4.4.10 from PyPI (with the hlapi.asyncio backend) to collect information from printers, and found that we're able to receive individual get responses for some devices, but not BULK get responses, which time out. For other devices, we're able to get all of them without issue.

I've captured traffic for the target devices from my machine, and found that the responses come in to the machine, but somehow aren't recognized as responses to the requests that were sent. I can also hit these same OIDs with `snmpbulkget` at my Ubuntu command line.

I've added logging statements to see that we're getting to right before our `bulkCmd()` call, which raises a timeout error.

What should I do next for troubleshooting on this? Is there a better place to ask this?

**Comment by daniel-fackrell-vasion** [comment]
Here are two network traces, one showing a bulk get request (5) and response (6) that pysnmp caught, and the other showing a bulk get request (5) and response (6) that pysnmp missed and continued waiting for.
[troubleshooting.zip](https://github.com/pysnmp/pysnmp/files/10403728/troubleshooting.zip)

**Comment by omrozowicz-splunk** [comment]
Can you please let me know what means that you use `4.4.10` version?For this repo versioning starts from `5.0.1`: https://pypi.org/project/pysnmplib/#history

**Comment by daniel-fackrell-vasion** [comment]
We've been using etingof/pysnmp, from which this repository was forked. I was trying to figure out where maintenance will be going forward and where to get support. I later found https://github.com/etingof/pysnmp/issues/429 and https://github.com/lextudio/pysnmp

I can re-post this as an issue there if that's more appropriate.

**Reply by lextm** [reply]
Already engaged in https://github.com/lextudio/pysnmp/issues/3
